File: src/backend/NHP_Domain/Entities/Logement.cs
namespace NHP_Domain.Entities;

public class Logement
{
    public long Id { get; set; }
    public String NomLogement { get; set; }
    public long PrixLogement { get; set; }
    public String DescriptionLogement { get; set; }
    
    // ManyToOne
    public virtual Image? Image { get; set; }
    public virtual TypeDestination? Destination { get; set; }
    public virtual Adresse? Adresse { get; set; }

    public override string ToString()
    {
        return $"nomLogement: {NomLogement} " +
               $"prix: {PrixLogement} " +
               $"descriptionLogement: {DescriptionLogement} " +
               $"urlImage: {Image} " +
               $"typeDestination: {Destination} " +
               $"adresse: {Adresse}";
    }
}

File: src/frontend/test/src/domain/entities/Logement.ts
import type { IImage } from '@/domain/entities/Image';
import type { IAdresse } from '@/domain/entities/Adresse';
import type { IAvis } from '@/domain/entities/Avis';
import type { IPrestationProposee } from '@/domain/entities/PrestationProposee';
import type { Destination } from '@/domain/entities/Destination';

export interface ILogement {
    id: number;
    nomLogement: string;
    prixLogement: number;
    descriptionLogement: string;
    image?: IImage;
    adresse?: IAdresse;
    prestationProposes?: IPrestationProposee[];
    destination: Destination;
    reduction?: number;
}

export class Logement implements ILogement {
    constructor(
        public id: number,
        public nomLogement: string,
        public prixLogement: number,
        public descriptionLogement: string,
        public destination: Destination,
        public image?: IImage,
        public adresse?: IAdresse,
        public prestationProposes?: IPrestationProposee[],
        public reduction?: number
    ) {}
}